src/perl/Permabit/FileSystem.pm

```diff
@@ -133,7 +133,20 @@ sub fsck {
   my ($self) = assertNumArgs(1, @_);
   my $machine = $self->{machine};
   my $devPath = $self->{device}->getSymbolicPath();
-  $machine->runSystemCmd("sudo fsck -t $self->{fsType} -a $devPath");
+  my $fsckCommand = "sudo fsck -t $self->{fsType} -a $devPath";
+  $log->debug("$machine->{hostname}: $fsckCommand");
+  my $errno = $machine->sendCommand($fsckCommand);
+  if ($errno > 1) {
+    confess("fsck failed on $machine->{hostname}: $errno\n"
+            . $machine->_getCmdOutput());
+  } elsif ($errno == 1) {
+    # Some "problems" like metadata in two different locations not
+    # getting updated together atomically can come up naturally and
+    # reasonably, and may be reported by fsck. (VDO-5638: "feature
+    # orphan_present is set but orphan file is clean")
+    $log->info("fsck fixed issues on $machine->{hostname}:\n"
```

Comment:
We probably don't need the ticket title; the ticket ID should be reference enough.